proposals/multiplatform/binary-signature.md

```diff
@@ -0,0 +1,601 @@
+@BinarySignatureName("doSomethingString")
+fun doSomething(param: Int): String { /* ... */ }
+```
+
+#### SinceKotlin/DeprecatedSinceKotlin
+
+The Kotlin standard library uses `@SinceKotlin("...")` and `@DeprecatedSinceKotlin(hiddenSince = "...")` to hide 
+declarations conditionally, based on the current API version in use (the value of the `-api-version` argument 
+when compiling the module where the usage is located). For the purposes of **call resolution** check, 
+they will taken into account, too:
+
+* Declarations with `@SinceKotlin(version)`, when the current API version is below the specified version, are excluded from the check.
+* Declarations with `@DeprecatedSinceKotlin(hiddenSince = version)`, when the current API version is the same or more recent as the version, are excluded from the check.
+
+### Override matching
+
+Override matching is performed using [**override matching signature**](#override-matching-signature). 
+It has to match between the override and the overridden method, which means the the source name, the binary signature name, 
+and all the parameter names must match or there will be a compilation error. In particular, this example is an error 
+(the source name does not match):
+
+```kotlin
+open class Base { 
+    open fun foo()
+}
+
+class Derived : Base() {
+    @SingatureName("foo")
+    override fun bar() // ERROR: 'bar' does not override anything
+}
+```
+
+And this example is an error (the signature name does not match):
+
+```kotlin
+open class Base { 
+    open fun foo()
+}
+
+class Derived : Base() {
+    @SingatureName("bar")
+    override fun foo() // ERROR: 'foo' does not override anything
+}
+```
+
+#### Overrides and properties
+
+Properties can have signature names on their getters and/or setters.
+
+* When overriding a read-only (`val`) property, the signature name for the getter must match.
+* When overriding a mutable (`var`) property, the signature name for BOTH the getter and the setter must match.
+
+#### Objective-C overrides in Kotlin/Native
+
+Objective-C has relaxed rules on overrides. In particular, it is possible to change the parameter types in Object-C on override. 
+These relaxed checks are implemented only in Kotlin/Native for Kotlin code that is produced as a result of ObjC-interop tooling and 
+their actual specifications are out of the scope of this KEEP.
+
+### Expect/actual matching
+
+Expect/actual matching checks both the source name of the declaration and the value of `@BinarySignatureName` 
+annotation just like the [override matching](#override-matching) does.
+
+### Declaring functions that differ only in parameter names
+
+In general, it is not in Kotlin style to have functions that different only in parameter name like it is customary
+in Objective-C and Swift, for example. In Kotlin you'd give these functions different names. However, it is conceivable
+that some domain-specific DSLs might want to have this kind of feature. The following code does not compile:
```

Comment:
There is phrase "domain-specific DSLs" (the 3rd sentence) which is equal to "domain-specific domain-specific languages". Is the doubling a typo or intended?

Reply:
Typo. Should be just `DSLs`.